common/types/head_tracker.go

```diff
@@ -24,6 +24,10 @@ type HeadTracker[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
 //
 //go:generate mockery --quiet --name HeadTrackable --output ./mocks/ --case=underscore
 type HeadTrackable[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
+	// OnNewLongestChain sends a new head when it becomes available. Subscribers can recursively trace the parent
+	// of the head to the EVM finality depth back. If this is not possible (e.g. due to recent boot, backfill not complete
```

Comment:
since this would be generalised, maybe we could replace the term `EVM`? or remove it?

Reply:
done